python/Ganga/Utility/execute.py

```diff
@@ -98,84 +188,74 @@ def execute(command,
             return_code=None):
     """
     Execute an external command.
+    This will execute an external python command when shell=False or an external bash command when shell=True
+    Args:
+        command (str): This is the command that we want to execute in string format
+        timeout (int): This is the timeout which we want to assign to a function and it will be killed if it runs for longer than n seconds
+        env (dict): This is the environment to use for launching the new command
+        cwd (str): This is the cwd the command is to be executed within.
+        shell (bool): True for a bash command to be executed, False for a command to be executed within Python
+        python_setup (str): A python command to be executed beore the main command is
+        eval_includes (str): An string used to construct an environment which, if passed, is used to eval the stdout into a python object
+        update_env (bool): Should we update the env being passed to what the env was after the command finished running
+        return_code (int): This is the returned code from the command which is executed
     """
+    shouldDeleteCWD=False
+    if cwd is None:
+        shouldDeteCWD=True
```

Comment:
Yes, I wasn't wanting to run commands which touch disk on AFS with this. They should be run in a temp directory which on many systems is faster.